ndcube/extra_coords/lookup_table_coord.py

```diff
@@ -0,0 +1,317 @@
+            name = "PixelFrame"
+            axes_type = "PIXEL"
+
+        axes_type = tuple([axes_type] * naxes)
+
+        return cf.CoordinateFrame(naxes, axes_type, axes_order, unit=unit,
+                                  axes_names=names, name=name, axis_physical_types=physical_types)
+
+    def _from_time(self, lookup_tables, mesh=False, names=None, physical_types=None, **kwargs):
+        if len(lookup_tables) > 1:
+            raise ValueError("Can only parse one time lookup table.")
+
+        time = lookup_tables[0]
+        deltas = (time[1:] - time[0]).to(u.s)
+        deltas = deltas.insert(0, 0)
+
+        def _generate_time_lookup(deltas):
+            return self._model_from_quantity((deltas,))
+
+        frame = cf.TemporalFrame(lookup_tables[0][0], unit=u.s, axes_names=names, name="TemporalFrame")
+        return DelayedLookupTable(deltas, _generate_time_lookup, mesh), frame
+
+    def _from_skycoord(self, lookup_tables, mesh=False, names=None, physical_types=None, **kwargs):
+        if len(lookup_tables) > 1:
+            raise ValueError("Can only parse one SkyCoord lookup table.")
+
+        def _generate_skycoord_lookup(components):
+            return self._model_from_quantity(components, mesh=mesh)
+
+        sc = lookup_tables[0]
+        components = tuple(getattr(sc.data, comp) for comp in sc.data.components)
+        ref_frame = sc.frame.replicate_without_data()
+        units = list(c.unit for c in components)
+
+        # TODO: Currently this limits you to 2D due to gwcs#120
+        frame = cf.CelestialFrame(reference_frame=ref_frame,
+                                  unit=units,
+                                  axes_names=names,
+                                  axis_physical_types=physical_types,
+                                  name="CelestialFrame")
+
+        return DelayedLookupTable(components, _generate_skycoord_lookup, mesh), frame
+
+    def _model_from_quantity(self, lookup_tables, mesh=False):
+        if len(lookup_tables) > 1:
+            return self._generate_compound_model(*lookup_tables, mesh=mesh)
+
+        return self.generate_tabular(lookup_tables[0])
+
+    def _from_quantity(self, lookup_tables, mesh=False, names=None, physical_types=None):
+        if not all(lt.unit.is_equivalent(lookup_tables[0].unit) for lt in lookup_tables):
+            raise u.UnitsError("All lookup tables must have equivalent units.")
+
+        unit = u.Quantity(lookup_tables).unit
+
+        frame = self._generate_generic_frame(len(lookup_tables), unit, names, physical_types)
+
+        return DelayedLookupTable(lookup_tables, partial(self._model_from_quantity, mesh=mesh), mesh), frame
+
+
+class DelayedLookupTable:
+    """
+    A wrapper to create a lookup table model on demand.
+    """
+
+    def __init__(self, lookup_table, model_function, mesh=True):
+        self.lookup_table = lookup_table
+        self.model_function = model_function
+        self.mesh = mesh
+
+    def __call__(self):
```

Comment:
Should``` mesh``` be incorporated in this function?  Or is it assumed that it is already accounted for because ```model_function``` is always a ``partial``` function that sets ```mesh``` and ```self.mesh``` is simply a record of that value?

Reply:
mesh is passed to the constructor of the class, so that when call is called the information about if mesh is being used is already in the class state.